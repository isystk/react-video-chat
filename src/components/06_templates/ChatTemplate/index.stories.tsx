import { Meta, Story } from '@storybook/react'
import React from 'react'
import ChatTemplate from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import ChanelService from '@/services/Chanel'

export default {
  title: '06_templates/ChatTemplate',
  component: ChatTemplate,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  main.setName('isystk')
  main.setRoomId('test')
  main.addChanel(
    new ChanelService(
      main,
      'all',
      'すべて',
      'all',
      'images/friends/Alpha_Team.png',
      'ルーム内のすべてのメンバー'
    )
  )
  main.setChanelId('all')

  return (
    <Context.Provider value={main}>
      <ChatTemplate />
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
