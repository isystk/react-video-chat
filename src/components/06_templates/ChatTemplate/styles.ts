import { css, keyframes } from '@emotion/css'

export const chatTemplate = css``
