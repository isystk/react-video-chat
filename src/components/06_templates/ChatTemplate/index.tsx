import Grid from '@mui/material/Grid'
import React, { useEffect, FC, useState } from 'react'
import { useRouter } from 'next/router'
import ChatMessages from '@/components/03_molecules/ChatMessages'
import InputFormChat from '@/components/03_molecules/InputFormChat'
import RecorderModal from '@/components/04_organisms/RecorderModal'
import { Url } from '@/constants/url'
import ChanelList from '@/components/03_molecules/ChanelList'
import ChanelInfo from '@/components/03_molecules/ChanelInfo'
import SendCallModal from '@/components/04_organisms/SendCallModal'
import ReceiveCallModal from '@/components/04_organisms/ReceiveCallModal'
import VideoTemplate from '@/components/06_templates/VideoTemplate'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { connect } from '@/components/hoc'
import HtmlSkeleton, { Title } from '@/components/05_layouts/HtmlSkeleton'
import * as _ from 'lodash'

/** ChatTemplateProps Props */
export type ChatTemplateProps = WithChildren & { main }
/** Presenter Props */
export type PresenterProps = ChatTemplateProps & { main; classes; windowHeight }

/** Presenter Component */
const ChatTemplatePresenter: FC<PresenterProps> = ({
  main,
  classes,
  windowHeight,
  appStyle,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{main.room.name}</Title>
    <div style={appStyle(windowHeight)}>
      <Grid container spacing={0}>
        <Grid item {...{ xs: 12, md: 3 }}>
          <div className="sp-hide">
            <ChanelList />
          </div>
        </Grid>
        <Grid item {...{ xs: 12, md: 6 }}>
          <ChatMessages />
          <InputFormChat />
        </Grid>
        <Grid item {...{ xs: 12, md: 3 }} style={{ backgroundColor: '#fff' }}>
          <div className="sp-hide">
            <ChanelInfo />
          </div>
        </Grid>
      </Grid>
      <RecorderModal />
      {main.video.nowCallSending && <SendCallModal />}
      {main.video.nowCallReceiving && <ReceiveCallModal />}
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const ChatTemplateContainer: React.FC<
  ContainerProps<ChatTemplateProps, PresenterProps>
> = ({ presenter, children, main, ...props }) => {
  const classes = useStyles()
  const router = useRouter()
  const [windowHeight, setWindowHeight] = useState(0)

  useEffect(() => {
    setWindowHeight(window.innerHeight)
    if (process.env.USE_AWS_AMPLIFY) {
      main.room.readRooms()
    }
  }, [])

  useEffect(() => {
    if (process.env.USE_AWS_AMPLIFY) {
      if (_.size(main.room.rooms) === 0) {
        return
      }
      // idがqueryで利用可能になったら処理される
      if (router.asPath !== router.route) {
        main.room.setRoomId(router.query.id + '')
      }
    } else {
      // idがqueryで利用可能になったら処理される
      if (router.asPath !== router.route) {
        main.room.setRoomName(router.query.id + '')
      }
    }
  }, [router, main.room.rooms])

  useEffect(() => {
    if (main.self.name === '') {
      router.push(Url.TOP)
    }
  }, [main.self.name])

  useEffect(() => {
    if (main.self.name !== '' && main.room.name !== '') {
      ;(async () => {
        // await main.mediaDevice.setMediaStream()
        await main.join()
      })()
    }
  }, [main.self.name, main.room.name])

  if (main.self.name === '') return <></>
  if (main.room.name === '') return <></>
  if (main.video.isPeerConnected) return <VideoTemplate main={main} />

  const appStyle = (vh) => {
    return {
      height: vh - 64,
    }
  }
  return presenter({
    children,
    main,
    classes,
    windowHeight,
    appStyle,
    ...props,
  })
}

export default connect<ChatTemplateProps, PresenterProps>(
  'ChatTemplate',
  ChatTemplatePresenter,
  ChatTemplateContainer
)
