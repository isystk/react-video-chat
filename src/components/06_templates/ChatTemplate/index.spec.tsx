import React from 'react'
import renderer from 'react-test-renderer'
import ChatTemplate, { ChatTemplateProps } from './index'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'
import ChanelService from '@/services/Chanel'
import { Provider } from 'react-redux'
import { store } from '@/stores'

describe('ChatTemplate', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.setRoomId('test')
    main.addChanel(
      new ChanelService(
        main,
        'all',
        'すべて',
        'all',
        'images/friends/Alpha_Team.png',
        'ルーム内のすべてのメンバー'
      )
    )
    main.setChanelId('all')
    const props: ChatTemplateProps = { main }
    const component = renderer.create(
      <Provider store={store}>
        <ChatTemplate {...props} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
