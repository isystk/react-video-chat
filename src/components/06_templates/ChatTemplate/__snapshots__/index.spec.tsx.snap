// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatTemplate Match Snapshot 1`] = `
<div
  className="area"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
    >
      <div
        className="sp-hide"
        style={
          Object {
            "height": -64,
          }
        }
      >
        <div
          className="chat_list"
          id="myChatList"
        >
          <div
            className="chatListTag active"
            onClick={[Function]}
          >
            <div
              className="head"
            >
              <img
                alt=""
                src="images/friends/Alpha_Team.png"
              />
            </div>
            <div
              className="mytext"
            >
              <div
                className="name"
              >
                すべて
              </div>
              <div
                className="dec"
              >
                ルーム内のすべてのメンバー
              </div>
            </div>
            <div
              className="msg_num"
            >
              7
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        className="chat_box"
      >
        <div
          className="chat_message"
          id="chatRoom"
          style={
            Object {
              "height": -133,
              "overflowY": "scroll",
            }
          }
        />
        <div
          className="makeStyles-chatInput-1"
        >
          <div
            className="makeStyles-myIcon2-3"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-colorAction"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
              />
            </svg>
          </div>
          <div
            className="makeStyles-myIcon2-3"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorAction"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                />
              </svg>
            </a>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-sendMsg-5 MuiFormControl-fullWidth"
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onKeyDown={[Function]}
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
            >
              メッセージを入力してください
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                name="message"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    メッセージを入力してください
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    >
      <div
        className="sp-hide"
      >
        <div
          className="me_block"
        >
          <div
            className="myHeadPhoto"
          >
            <img
              alt=""
              src="images/friends/Alpha_Team.png"
            />
          </div>
          <div
            className="myName"
          >
            すべて
          </div>
          <div
            className="myTitle"
          >
            ルーム内のすべてのメンバー
          </div>
          <div
            className="myVideo"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 6.5l-4 4V7c0-.55-.45-1-1-1H9.82L21 17.18V6.5zM3.27 2L2 3.27 4.73 6H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.21 0 .39-.08.54-.18L19.73 21 21 19.73 3.27 2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
